collector/src/main/java/com/bitmovin/analytics/retryBackend/RetryQueue.kt

```diff
@@ -0,0 +1,84 @@
+package com.bitmovin.analytics.retryBackend
+
+import android.util.Log
+import java.util.Calendar
+import java.util.Date
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.Comparator
+import kotlin.math.pow
+
+class RetryQueue {
+    private val TAG = "RetryQueue"
+    private val lock = ReentrantLock()
+    private val MAX_RETRY_TIME = 300 // in seconds
+    private val MAX_RETRY_SAMPLES = 100
+    private val MAX_BACKOFF_INTERVAL = 64
+
+    fun getMaxSampleNumber() = MAX_RETRY_SAMPLES
```

Comment:
why does this need to be a fun and not just a private val?

Reply:
I did it like this so I can change it to some smaller value in tests.
Other option I can think of is to use `public var MAX_BACKOFF_INTERVAL` and then change it 🤔

Reply:
ok no it's fine this way - just looked odd